### PR TITLE
[Xamarin.Android.Build.Tests] Add string password keystore tests

### PR DESCRIPTION
**Pull request by dellis1972**
Context #4036
    
Add unit tests to make sure we can use a strong password with the keytool and signing tools.

### CODE REVIEW

**Comment by jonpryor** [comment]
@dellis1972: I don't understand the interaction between this PR and https://github.com/xamarin/xamarin-android/issues/4036#issuecomment-607830864 .

Specifically, the comment on Issue #4036 implies that the Android+Java toolchain doesn't like "certain characters" such as `@`.  Yet this PR uses a password that contains `@`, *it works*, and I don't see any relevant changes in the PR which would cause it to work with this PR but not before this PR.

What am I missing?

**Comment by dellis1972** [comment]
@jonpryor It seems its not very consistent. See https://github.com/xamarin/xamarin-android/issues/4036#issuecomment-613424371. With certain characters I get

```
keytool -genkeypair -alias teststringkey -storepass "r@^]$test12" -keypass "r@^]$test12" -keystore foo.keystore -dname "CN=Android Debug,O=Android,C=US" -keyalg RSA -validity 10000 -storetype pkcs12
keytool error: java.lang.Exception: Key password must be at least 6 characters
```

BUT those same characters work fine in the unit test. I have a feeling keytool is a bit weird.

Anyway I updated the PR to just say we are adding a strong password unit test.

File: src/Xamarin.Android.Build.Tasks/Tests/Xamarin.Android.Build.Tests/Tasks/KeyToolTests.cs [hunk 1]
```diff
@@ -163,5 +165,47 @@ public void CreateDebugKeyStore ()
...
+				KeyPass = pass,
+				Command = "-list",
+				ToolPath = keyToolPath,
+			};
+
+			Assert.IsTrue (keyToolTask.Execute (), "Task should have succeeded.");
+			Assert.AreEqual (0, errors.Count, "Task should have no errors.");
+			Assert.AreEqual (0, warnings.Count, "Task should have no warnings.");
+			Assert.AreEqual (0, task.ExitCode, "ExitCode should have been 0");
+			Assert.IsTrue (messages.Select (x => x.Message).Contains ("Certificate fingerprint (SHA1):"), "Certificate SHA1 should have been printed.");
```
Comment (jonpryor) [inline src/Xamarin.Android.Build.Tasks/Tests/Xamarin.Android.Build.Tests/Tasks/KeyToolTests.cs:208]: This assert is failing: https://devdiv.visualstudio.com/DevDiv/_build/results?buildId=3640507&view=ms.vss-test-web.build-test-results-tab&runId=12594914&resultId=100127&paneView=debug

```
Certificate SHA1 should have been printed.
 Expected: True
 But was: False
```

Comment (dellis1972) [reply src/Xamarin.Android.Build.Tasks/Tests/Xamarin.Android.Build.Tests/Tasks/KeyToolTests.cs:208]: Weird cos its in the output `Certificate fingerprint (SHA1):` ... I'll take a look

**Pull request merged**